src/main/java/net/dv8tion/jda/internal/utils/PermissionUtil.java

```diff
@@ -314,7 +314,8 @@ public static long getEffectivePermission(Member member)
             if (isApplied(permission, Permission.ADMINISTRATOR.getRawValue()))
                 return Permission.ALL_PERMISSIONS;
         }
-
+        if (member.isTimedOut())
+            permission &= Permission.getRaw(Permission.VIEW_CHANNEL, Permission.MESSAGE_HISTORY);
```

Comment:
I think you also have to check this for channel perms

Reply:
that seems to use the member perms. https://github.com/DV8FromTheWorld/JDA/blob/master/src/main/java/net/dv8tion/jda/internal/utils/PermissionUtil.java#L353

Reply:
Yes, as a base. But overwrites add more permissions afterwards. And from my understanding, those permissions are also disabled by timeouts.